File: 2016/python/day15/part2.py
#!/usr/bin/python


def parse(disc):
    return {'position': int(disc[11]), 'size': int(disc[3])}


with open('input.txt') as file_handler:
    discs = [parse(line.rstrip('.\n').split()) for line in file_handler]

discs.append({'position': 0, 'size': 11})
time = 0

while True:
    for index, disc in enumerate(discs):
        if (disc['position'] + index + time + 1) % disc['size'] != 0:
            break
    else:
        print time
        break

    time += 1

File: 2015/ruby/day16/part2.rb
#!/usr/bin/ruby

def compare(type, goalVal, testVal)
    if type == 'cats' or type == 'trees'
        if testVal > goalVal
            return true
        end
        return false
    elsif type == 'pomeranians' or type == 'goldfish'
        if testVal < goalVal
            return true
        end
        return false
    elsif testVal == goalVal
        return true
    end
    return false
end

begin
    File.open('input.txt', 'r') do |file|
        goal = {
            'children' => 3,
            'cats' => 7,
            'samoyeds' => 2,
            'pomeranians' => 3,
            'akitas' => 0,
            'vizslas' => 0,
            'goldfish' => 5,
            'trees' => 3,
            'cars' => 2,
            'perfumes' => 1
        }

        file.each_line do |line|
            stats = line.delete(':,').downcase.split(' ')

            if (compare(stats[2], goal[stats[2]], stats[3].to_i) and
                compare(stats[4], goal[stats[4]], stats[5].to_i) and
                compare(stats[6], goal[stats[6]], stats[7].to_i))
                puts stats[1]
                break
            end
        end
    end
rescue SystemCallError
    puts 'input.txt not found.'
end
